packages/core/src/hooks/form/index.ts

```diff
@@ -1 +1,341 @@
+
+import {
+  redirectPage,
+  deferExecution,
+  pickNotDeprecated,
+} from "@definitions/helpers";
+
+import type { BaseRecord, HttpError, BaseKey } from "../../interfaces";
+import type { UpdateResponse, CreateResponse } from "../../interfaces";
+import type { UpdateParams } from "../data/useUpdate";
+import type { UseCreateParams } from "../data/useCreate";
+import type { UseFormProps, UseFormReturnType } from "./types";
+
+export type { ActionParams, UseFormProps, UseFormReturnType } from "./types";
+
+/**
+ * This hook orchestrates Refine's data hooks to create, edit, and clone data. It also provides a set of features to make it easier for users to implement their real world needs and handle edge cases such as redirects, invalidation, auto-save and more.
+ *
+ * @see {@link https://refine.dev/docs/data/hooks/use-form} for more details.
+ *
+ * @typeParam TQueryFnData - Result data returned by the query function. Extends {@link https://refine.dev/docs/core/interface-references/#baserecord `BaseRecord`}
+ * @typeParam TError - Custom error object that extends {@link https://refine.dev/docs/core/interface-references/#httperror `HttpError`}
+ * @typeParam TVariables - Values for params. default `{}`
+ * @typeParam TData - Result data returned by the `select` function. Extends {@link https://refine.dev/docs/core/interface-references/#baserecord `BaseRecord`}. Defaults to `TQueryFnData`
+ * @typeParam TResponse - Result data returned by the mutation function. Extends {@link https://refine.dev/docs/core/interface-references/#baserecord `BaseRecord`}. Defaults to `TData`
+ * @typeParam TResponseError - Custom error object that extends {@link https://refine.dev/docs/core/interface-references/#httperror `HttpError`}. Defaults to `TError`
+ *
+ */
+export const useForm = <
+  TQueryFnData extends BaseRecord = BaseRecord,
+  TError extends HttpError = HttpError,
+  TVariables = {},
+  TData extends BaseRecord = TQueryFnData,
+  TResponse extends BaseRecord = TData,
+  TResponseError extends HttpError = TError,
+>(
+  props: UseFormProps<
+    TQueryFnData,
+    TError,
+    TVariables,
+    TData,
+    TResponse,
+    TResponseError
+  > = {},
+): UseFormReturnType<
+  TQueryFnData,
+  TError,
+  TVariables,
+  TData,
+  TResponse,
+  TResponseError
+> => {
+  const getMeta = useMeta();
+  const invalidate = useInvalidate();
+  const { redirect: defaultRedirect } = useRefineOptions();
+  const { mutationMode: defaultMutationMode } = useMutationMode();
+
+  const { setWarnWhen } = useWarnAboutChange();
+  const handleSubmitWithRedirect = useRedirectionAfterSubmission();
+
+  const pickedMeta = pickNotDeprecated(props.meta, props.metaData);
+  const mutationMode = props.mutationMode ?? defaultMutationMode;
+
+  const {
+    resource,
+    action: routeAction,
+    identifier,
+  } = useResource(props.resource);
+
+  const [autosaved, setAutosaved] = React.useState(false);
+
+  const [id, setId] = useId({ id: props.id, resource: props.resource });
```

Comment:
```suggestion
  const [id, setId] = useId({ id: props.id, resource: props.resource });
```

Do we handle the case, `resource` string is not the name of the string, but it's `identifier` of the resource?